src/components/Identity/createComponent.js

```diff
@@ -74,61 +56,54 @@ export default (idSyncs, config, logger, cookieJar, optIn, eventManager) => {
       // TO-DOCUMENT: We wait for ECID before trigger any events.
       onBeforeDataCollection({ payload }) {
         return optIn.whenOptedIn().then(() => {
-          const ecid = getEcid();
-
           let promise;
 
-          if (ecid) {
-            addIdsContext(payload, ecid);
-          } else if (deferredForEcid) {
-            // We don't have an ECID, but the first request has gone out to
-            // fetch it. We must wait for the response to come back with the
-            // ECID before we can apply it to this payload.
-            logger.log("Delaying request while retrieving ECID from server.");
-            promise = deferredForEcid.promise.then(() => {
-              logger.log("Resuming previously delayed request.");
-              addIdsContext(payload, getEcid());
-            });
-          } else {
-            // We don't have an ECID and no request has gone out to fetch it.
-            // We won't apply the ECID to this request, but we'll set up a
-            // promise so that future requests can know when the ECID has returned.
-            deferredForEcid = defer();
-            payload.expectResponse();
-            if (
-              config.thirdPartyCookiesEnabled &&
-              areThirdPartyCookiesSupported(getBrowser(window))
-            ) {
-              payload.useIdThirdPartyDomain();
+          if (!hasEcid()) {
+            if (deferredForEcid) {
+              // We don't have an ECID, but the first request has gone out to
+              // fetch it. We must wait for the response to come back with the
+              // ECID before we can apply it to this payload.
+              logger.log("Delaying request while retrieving ECID from server.");
+              promise = deferredForEcid.promise.then(() => {
+                logger.log("Resuming previously delayed request.");
+              });
+            } else {
+              // We don't have an ECID and no request has gone out to fetch it.
+              // We won't apply the ECID to this request, but we'll set up a
+              // promise so that future requests can know when the ECID has returned.
+              deferredForEcid = defer();
+              payload.expectResponse();
+              // If third-party cookies are enabled by the customer and
+              // supported by the browser, we will send the request to a
+              // a third-party domain that allows for more accurate
+              // identification of the user through use of a third-party cookie.
+              if (
+                config.thirdPartyCookiesEnabled &&
+                areThirdPartyCookiesSupported(getBrowser(window))
+              ) {
+                payload.useIdThirdPartyDomain();
```

Comment:
I'd love to be able to track this in the context data to be able to answer these questions:
1. What percentage of users did we attempt to call the third party domain?
2. What percentage of those call successfully set the third part cookie?
3. I'd like to create a segment for those using browsers where we tried to use the third party domain.
4. What percentage of new users already had an ecid?

I wonder if we should track as part of context the edge domain we called?  We could also track whether the third party cookie was set.